slack-plugin/src/main/kotlin/slack/gradle/SlackExtension.kt

```diff
@@ -636,12 +645,73 @@ public abstract class DaggerHandler @Inject constructor(objects: ObjectFactory)
 
 @SlackExtensionMarker
 @Suppress("UnnecessaryAbstractClass")
-public abstract class AndroidHandler @Inject constructor(objects: ObjectFactory) {
+public abstract class ComposeHandler
+@Inject
+constructor(
+  objects: ObjectFactory,
+  globalSlackProperties: SlackProperties,
+  slackProperties: SlackProperties,
+  versionCatalog: VersionCatalog
+) {
+
+  private val composeBundleAlias =
+    globalSlackProperties.defaultComposeAndroidBundleAlias?.let { alias ->
+      versionCatalog.findBundle(alias).orElse(null)
+    }
+  private val composeCompilerVersion by lazy {
+    slackProperties.versions.composeCompiler
+      ?: error("Missing `compose-compiler` version in catalog")
+  }
+  internal val enabled = objects.property<Boolean>().convention(false)
+
+  /**
+   * This is weird! Due to the non-property nature of the AGP buildFeatures and composeOptions DSLs,
+   * we can't lazily chain their values to our own extension's properties. Because of this, we
+   * lazily set this instance from [StandardProjectConfigurations] during Android extension
+   * evaluation and then make calls to [enable] directly set the values on this instance. Ideally we
+   * could eventually remove this if/when AGP finally makes these properties lazy.
+   */
+  internal var androidExtension: CommonExtension<*, *, *, *>? = null
```

Comment:
I have similar-ish code in Square's AGP anti-corruption layer (facade/wrapper). We actually instantiate the facade lazily and pass it in via the `@Inject`ed constructor.

Reply:
Nice. I couldn't find an easy way to make it constructor injected in our side for now